src/tools/admin/conseillers/fix-echec-rupture.js

```diff
@@ -0,0 +1,391 @@
+  'listeSubordonnes.type': 'conseillers',
+  'listeSubordonnes.liste': {
+    $elemMatch: { $eq: idCNFS }
+  } },
+{ $pull: { 'listeSubordonnes.liste': idCNFS } },
+);
+const getDoublon = db => async (idCNFS, emailPerso) => await db.collection('conseillers').findOne({
+  _id: { $ne: idCNFS },
+  email: emailPerso,
+  disponible: false
+});
+const updateDoublon = db => async (idCNFS, emailPerso) => {
+  await db.collection('conseillers').updateMany({
+    _id: { $ne: idCNFS },
+    email: emailPerso,
+  },
+  {
+    $set: {
+      disponible: true,
+    },
+  });
+  await db.collection('misesEnRelation').updateMany(
+    {
+      '_id': { $ne: idCNFS },
+      'conseillerObj.email': emailPerso,
+      'statut': 'finalisee_non_disponible',
+    },
+    { $set: { 'statut': 'nouvelle', 'conseillerObj.disponible': true } },
+  );
+};
+const conseillerRecruteReinscriptionCandidat = db => async userCandidatDoublon => {
+  await db.collection('users').deleteOne({ _id: userCandidatDoublon._id });
+  await db.collection('conseillers').updateOne(
+    { _id: userCandidatDoublon.entity.oid },
+    { $set: { userCreated: false } },
+  );
+  await db.collection('misesEnRelation').updateMany(
+    { 'conseiller.$id': userCandidatDoublon.entity.oid },
+    { $set: { 'conseillerObj.userCreated': false } }
+  );
+};
+const getUserCandidatDoublon = db => async conseiller => await db.collection('users').findOne({ roles: { $in: ['candidat'] }, name: conseiller.email });
+const getCoop = db => async idCNFS => await db.collection('users').findOne({ 'roles': { $in: ['conseiller'] }, 'entity.$id': idCNFS });
+const updateUserCompteCandidat = db => async (conseiller, getUserConseiller, userCandidatDoublon) => {
+  // nécessaire pour ne pas avoir d'erreur de duplicate key
+  const nameCandidat = getUserConseiller !== null && userCandidatDoublon === null ? { name: conseiller.email } : {};
+  await db.collection('users').updateOne(
+    { _id: getUserConseiller._id },
+    { $set: {
+      ...nameCandidat,
+      roles: ['candidat'],
+      token: uuidv4(),
+      tokenCreatedAt: new Date(),
+      mailSentDate: null, // pour le mécanisme de relance d'invitation candidat
+      passwordCreated: false,
+    } },
+  );
+};
+const updateMisesEnRelationRupture = db => async (idCNFS, idStructure, dateRupture, validateur) => await db.collection('misesEnRelation').updateOne(
+  {
+    'conseiller.$id': idCNFS,
+    'structure.$id': idStructure,
+    'statut': { '$in': ['nouvelle_rupture', 'finalisee', 'finalisee_rupture'] }
+  },
+  {
+    $set: {
+      statut: 'finalisee_rupture',
+      dateRupture: new Date(dateRupture),
+      validateurRupture: { email: validateur, date: new Date() },
+    },
+    $unset: {
+      dossierIncompletRupture: '',
+    },
+  },
+);
+const getMisesEnRelationNonDispo = db => async idCNFS => await db.collection('misesEnRelation').findOne(
+  {
+    'conseiller.$id': idCNFS,
+    'statut': 'finalisee_non_disponible',
```

Comment:
ca me fait penser dans l'import coop : il y a un souci sur les doublons 
            await db.collection('misesEnRelation').updateMany({
              'conseillerObj.idPG': { $ne: idPGConseiller },
              'conseillerObj.email': conseillerOriginal.email,
              'statut': { $ne: 'finalisee_rupture' }
            }, {
            
            
            Peut etre passer en white liste : { $in: ['nouvelle', 'interessee', 'nonInteressee', 'recrutee'] }

Reply:
Oui mise à jour à effectué également pour cette partie 👍 (côté pilotage également)

Reply:
Ah je viens de relire, la modif est faite ici => https://github.com/anct-cnum/api-conseiller-numerique/pull/1185/files#diff-fac702395e728fc16f7c995886ae999bc501327f69e74fcd236e3422c0879435